### ISSUE TITLE
No output when running phpcs

### ISSUE THREAD

**Issue by rilwis** [squizlabs/PHP_CodeSniffer#3880]
## Describe the bug

I installed PHPCS and some other rules for a local package, which is a WordPress plugin. And when I run `phpcs`, it doesn't output anything.

Here is my `composer.json` file:

```json
{
	"name": "elightup/slim-seo-link-manager",
	"type": "wordpress-plugin",
	"repositories": [
		{
			"type": "git",
			"url": "git@github.com:elightup/plugin-updater.git"
		},
		{
			"type": "git",
			"url": "git@github.com:elightup/plugin-search.git"
		}
	],
	"autoload": {
		"psr-4": {
			"SlimSEOPro\\LinkManager\\": "src/",
			"SlimSEO\\": "lib"
		}
	},
	"minimum-stability": "dev",
	"prefer-stable": true,
	"config": {
		"prepend-autoloader": false,
		"allow-plugins": {
			"dealerdirect/phpcodesniffer-composer-installer": true
		}
	},
	"extra": {
		"dev-files": {
			"elightup/plugin-updater": [
				"plugin-updater.php"
			],
			"elightup/plugin-search": [
				"plugin-search.php"
			],
			"elightup/*": [
				".git/",
				".gitignore",
				"composer.json",
				"phpcs.xml"
			]
		}
	},
	"require": {
		"elightup/plugin-updater": "dev-master",
		"deliciousbrains/wp-background-processing": "^1.0",
		"voku/stop-words": "^2.0",
		"elightup/plugin-search": "dev-master"
	},
	"require-dev": {
		"dealerdirect/phpcodesniffer-composer-installer": "^1.0",
		"wp-coding-standards/wpcs": "dev-develop",
		"phpcompatibility/phpcompatibility-wp": "^2.1"
	}
}
```

### Custom ruleset

Here is my `phpcs.xml` file:

```xml
<?xml version="1.0"?>
<ruleset name="eLightUp WordPress Coding Standards">
	<!-- See https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml -->
	<!-- See https://github.com/WordPress/WordPress-Coding-Standards -->
	<!-- See https://github.com/PHPCompatibility/PHPCompatibilityWP -->

	<!--
	#############################################################################
	COMMAND LINE ARGUMENTS
	https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml
	#############################################################################
	-->

	<!-- Pass some flags to PHPCS:
		 p flag: Show progress of the run.
		 s flag: Show sniff codes in all reports.
	-->
	<arg value="ps"/>


	<!-- Check up to 8 files simultaneously. -->
	<arg name="parallel" value="8"/>

	<!-- Check PHP files only. JavaScript and CSS files are checked separately using the @wordpress/scripts package. -->
	<arg name="extensions" value="php"/>

	<!-- Check all files in this directory and the directories below it. -->
	<file>.</file>

	<!-- Exclude patterns. -->
	<exclude-pattern>/vendor/*</exclude-pattern>
	<exclude-pattern>/tests/*</exclude-pattern>
	<exclude-pattern>/.github/*</exclude-pattern>

	<rule ref="WordPress">
		<!-- Use PSR-4 naming -->
		<exclude name="WordPress.Files.FileName.NotHyphenatedLowercase" />
		<exclude name="WordPress.Files.FileName.InvalidClassFileName" />

		<!-- Only comment when neccessary -->
		<exclude name="Squiz.Commenting.FileComment.Missing" />
		<exclude name="Squiz.Commenting.FileComment.MissingPackageTag" />
		<exclude name="Squiz.Commenting.ClassComment.Missing" />
		<exclude name="Squiz.Commenting.FunctionComment.Missing" />
		<exclude name="Squiz.Commenting.FunctionComment.MissingParamTag" />
		<exclude name="Squiz.Commenting.VariableComment.Missing" />
		<exclude name="Squiz.Commenting.InlineComment.InvalidEndChar" />

		<!-- Write shorter -->
		<exclude name="Universal.Arrays.DisallowShortArraySyntax.Found" />
		<exclude name="Generic.Arrays.DisallowShortArraySyntax.Found" />
		<exclude name="Generic.PHP.DisallowShortOpenTag.EchoFound" />
		<exclude name="Generic.Commenting.DocComment.SpacingBeforeTags" />
		<exclude name="Squiz.PHP.EmbeddedPhp.NoSemicolon" />
		<exclude name="WordPress.Classes.ClassInstantiation.MissingParenthesis" />
		<exclude name="Universal.Operators.DisallowShortTernary.Found" />
		<exclude name="WordPress.PHP.DisallowShortTernary.Found" />
		<exclude name="PSR12.Classes.ClassInstantiation.MissingParentheses" />

		<!-- Optional -->
		<exclude name="PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket" />
		<exclude name="PEAR.Functions.FunctionCallSignature.CloseBracketLine" />
		<exclude name="PEAR.Functions.FunctionCallSignature.MultipleArguments" />
		<exclude name="WordPress.PHP.YodaConditions.NotYoda" />
	</rule>

	<rule ref="Generic.Arrays.DisallowLongArraySyntax" />

	<!-- Set the minimum supported WP version. This is used by several sniffs.
		 The minimum version set here should be in line with the minimum WP version
		 as set in the "Requires at least" tag in the readme.txt file. -->
	<config name="minimum_supported_wp_version" value="5.9"/>

	<rule ref="WordPress.WP.I18n">
		<properties>
			<property name="text_domain" type="array">
				<element value="slim-seo-link-manager"/>
				<element value="slim-seo"/>
			</property>
		</properties>
	</rule>

	<rule ref="WordPress.NamingConventions.PrefixAllGlobals">
		<properties>
			<property name="prefixes" type="array">
				<element value="SlimSEOPro\LinkManager"/>
				<element value="SlimSEO"/>
				<element value="slim_seo_link_manager"/>
				<element value="slim_seo"/>
			</property>
		</properties>
	</rule>

	<!--
	#############################################################################
	USE THE PHPCompatibility RULESET
	#############################################################################
	-->
	<config name="testVersion" value="7.2-"/>
	<rule ref="PHPCompatibilityWP"/>
</ruleset>
```

### To reproduce

After installing dependencies, when I run the following commands nothing is outputted in the console:

```bash
./vendor/bin/phpcs -v
./vendor/bin/phpcs -h
./vendor/bin/phpcs -p .
```

I also run them with variations like setting the `--standard="phpcs.xml"` or set the folder to `./src/`, etc. - still no luck.

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | MacOS Big Sur 11.5 (chip Apple M1)
| PHP version | 8.2.1
| PHP_CodeSniffer version | 3.8.0
| Standard | WordPress
| Install type | Composer local (works find with Composer global)

## Additional context

I tried to add some `die` to the `vendor/squizlabs/php_codesniffer` folder to debug, and found that the code stops at `autoload.php` file, on this line:

https://github.com/squizlabs/PHP_CodeSniffer/blob/master/autoload.php#L77

It seems to load Composer's autoload, and then stops.

I don't know how to debug further.

Please take a look.

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue. The issue #1502 is similar and is closed, and I was encouraged to open a new one.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@rilwis I've tried, but cannot reproduce your issue.

This is what I did to try and reproduce the issue:
* Created a directory `phpcs-3880`
* Put your `composer.json`in that directory.
* Put your config in that directory in a file called `phpcs.xml.dist`
* Added a simple `test.php` file with some code - `<?php echo $var;` - which should trigger an error based on your ruleset.
* Removed the references to the `elightup` repos from the `composer.json` as I don't have access to them.
* Ran `composer install`.
* Ran `vendor/bin/phpcs -i` and got the following output:
    ```
    The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend, PHPCompatibility, PHPCompatibilityParagonieRandomCompat, PHPCompatibilityParagonieSodiumCompat, PHPCompatibilityWP, Modernize, NormalizedArrays, Universal, PHPCSUtils, WordPress, WordPress-Core, WordPress-Docs and WordPress-Extra
    ```
* Ran `vendor/bin/phpcs - v` and got the following output:
```
# "vendor/bin/phpcs" -v
Registering sniffs in the eLightUp WordPress Coding Standards standard... DONE (298 sniffs registered)
Creating file list... DONE (1 files in queue)
Changing into directory path/to/phpcs-3880
Processing test.php [PHP => 7 tokens in 3 lines]... DONE in 360ms (1 errors, 0 warnings)

FILE: path/to/phpcs-3880/test.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 3 | ERROR | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$var'.
   |       | (WordPress.Security.EscapeOutput.OutputNotEscaped)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

In other words, without more information to allow someone to reproduce your issue, there is nothing anyone can do.

It might help to know what versions of various packages are installed and on which OS, but you removed the "Versions" table from the bug report template.


P.S.: [WordPressCS 3.0.0 has been released](https://make.wordpress.org/core/2023/08/21/wordpresscs-3-0-0-is-now-available/), so I suggest you update your `require-dev` to use `^3.0` for the `wp-coding-standards/wpcs` package.
More info in the [changelog](https://github.com/WordPress/WordPress-Coding-Standards/releases/tag/3.0.0) and [upgrade guide](https://github.com/WordPress/WordPress-Coding-Standards/wiki/Upgrade-Guide-to-WordPressCS-3.0.0-for-ruleset-maintainers)
.

**Comment by rilwis** [comment]
@jrfnl 

That's strange.

When I replicate your test with the same steps, I see it works as you described.

However, when I check with my code, I can't find anything wrong. I recorded a screencast so you can see what I'm doing:

https://mega.nz/file/X6w0BZQR#8z7GAtfTPlxpWTcQDWIEKrFIXKafcyGMaaE28mvkaPg

I also updated the issue with "Versions" section for more details.

**Comment by fredden** [comment]
@rilwis I can see that you've listed the PHP_CodeSniffer version as `3.8.0`, but that version does not exist currently: https://github.com/squizlabs/PHP_CodeSniffer/tags. Please can you confirm what version you're using, and perhaps how you've installed PHP_CodeSniffer? I don't see this listed in the `composer.json` file supplied, so I have to assume that it's installed somewhere else on your system. If you install it directly into this project (with `composer require --dev squizlabs/php_codesniffer`), does this work as expected?

**Comment by fredden** [comment]
> When I replicate your test with the same steps, I see it works as you described.

The steps there include removing all the "elightup" packages. Might they be causing this issue? @rilwis please can you confirm if the bug exists with/without these packages?

**Comment by fredden** [comment]
I also noticed in the video that after modifying `composer.json` in your text editor, you only run `composer install` and don't actually re-install any files. There's a warning from Composer that its content-hash no longer matches. The "cleaned" files are probably still there; I'd recommend deleting all of `vendor/` when making changes such as that, before running `composer install` to get Composer to put the files in place.

**Comment by rilwis** [comment]
Sorry for the late reply. I've just back from a holiday.

Regarding the questions, I've recorded a new video to show you a whole process:

https://mega.nz/file/amIDVBYY#LTAmlDnEaiw9M89RvcMwE-_cuk1rQmGpp4h1jMKna1k

What I did:

- Clone the repo freshly from Github.
- Remove all private repos and `extra` rules (which were used to remove redundant files in `vendor` folder). So everything now looks like a normal Composer install.
- Run `composer install`. The `vendor` folder is now fresh and has proper files.
- Run `./vendor/bin/phpcs -v` and nothing appears.

Note that I have `phpcs` installed on my machine globally, so when I run `phpcs -v`, it outputs things correctly. Only the local version of `phpcs`, which is `./vendor/bin/phpcs`, doesn't output anything.

**Comment by fredden** [comment]
Thanks for the update @rilwis. I notice in this new video that you are still not installing the packages that you think you are. This is because you do not modify the `composer.lock` file when making changes to the `composer.json` file in your editor. There's a warning from Composer when you run `composer install` saying this too.

Please can you try again with this script. None of this requires you to leave the terminal / use your editor.

```sh
# set up
set -x
cd $(mktemp -d)
git clone git@github.com:elightup/slim-seo-link-manager.git .

# confirm bug is still present
composer install
vendor/bin/phpcs -i

# remove packages not available to the public
composer config --unset repo.1
composer config --unset repo.0
composer config --unset extra.dev-files
composer remove elightup/plugin-updater elightup/plugin-search

# test if bug still persists
vendor/bin/phpcs -i
vendor/bin/phpcs -v

# some diagnostics
head -n 1 vendor/bin/phpcs
type php
php --version
composer --version
composer show
composer global show
composer global depends composer-plugin-api
```

**Comment by rilwis** [comment]
@fredden 

Thanks for the scripts. Here are the screenshots:

![1](https://github.com/squizlabs/PHP_CodeSniffer/assets/668274/8932491c-87d5-4d24-98ce-d715b2644de3)

![2](https://github.com/squizlabs/PHP_CodeSniffer/assets/668274/3e4ae4ed-ea56-4cc5-a0dd-8a8b2c8d522e)

![3](https://github.com/squizlabs/PHP_CodeSniffer/assets/668274/4118a22b-ea85-4fe3-b9d9-58ba27210e27)

![4](https://github.com/squizlabs/PHP_CodeSniffer/assets/668274/6016e3af-d36b-4d58-9ccb-f67087f34049)

I guess it's working after removing packages. But don't know why.

**Comment by jrfnl** [comment]
@rilwis This sounds like it just confirms that this is not a PHPCS problem, but a problem with those one or both of those two (`private`) packages. I suggest you report your issue there.

**Comment by rilwis** [comment]
After testing again with some commands, I found that when I run:

```bash
composer remove elightup/plugin-search
```

Then the command `./vendor/bin/phpcs -i` works again.

However, when I require `elightup/plugin-search`, then it failed.

Digging a little deeper, in the repo `elightup/plugin-search`, the autoload rule loads a file `bootstrap.php`, which contains some code for WordPress specifically. When running `vendor/bin/phpcs -i`, this file is also loaded. And because this is not a WordPress installation, the `bootstrap.php` file causes an error and makes the script fails.

I think it can be mentioned somehow in the docs, that any autoload files must run properly in the CLI environment, in order to not break `phpcs`.

I think this issue can be closed now.

Thank you very much for your help!

**Comment by jrfnl** [comment]
> I think it can be mentioned somehow in the docs, that any autoload files must run properly in the CLI environment, in order to not break phpcs.

That is not just an issue for PHPCS, but for literally every tool out there. If you have a `files` autoload directive, it is a given that that file should _always_ run without side-effects.

**Comment by BrianHenryIE** [comment]
I had the same problem here. It was inside my own code being autoloaded with a files autoloader.

```json

    "autoload": {

        "files": [
            "lib/functions.php"
        ]
    },
```

I at the top of `functions.php` I had:

```php
if ( ! defined( 'ABSPATH' ) ) {
	exit;
}
```
which I changed to 
```php
if ( ! defined( 'ABSPATH' ) ) {
	return;
}
```

and things began to work again.

**Comment by hinnerk-a** [comment]
@BrianHenryIE Same fix here. Thanks a lot! Great job, made my day!
